session/session_test.go

```diff
@@ -3314,58 +3313,99 @@ PARTITION BY RANGE (c) (
+	result = tk.MustQuery("select * from t1")    // read dc-1 and dc-2 with global scope
 	c.Assert(len(result.Rows()), Equals, 1)
+
+	// begin and commit with global txn scope
 	tk.MustExec("begin")
 	txn, err := tk.Se.Txn(true)
 	c.Assert(err, IsNil)
 	c.Assert(tk.Se.GetSessionVars().TxnCtx.TxnScope, Equals, oracle.GlobalTxnScope)
 	c.Assert(txn.Valid(), IsTrue)
-	tk.MustExec("insert into t1 (c) values (1)") // in dc-1 with global scope
-	result = tk.MustQuery("select * from t1")
+	tk.MustExec("insert into t1 (c) values (1)") // write dc-1 with global scope
+	result = tk.MustQuery("select * from t1")    // read dc-1 and dc-2 with global scope
 	c.Assert(len(result.Rows()), Equals, 2)
 	c.Assert(txn.Valid(), IsTrue)
 	tk.MustExec("commit")
 	result = tk.MustQuery("select * from t1")
 	c.Assert(len(result.Rows()), Equals, 2)
-	tk.MustExec("insert into t1 (c) values (101)") // in dc-2 with global scope
+
+	// begin and rollback with global txn scope
+	tk.MustExec("begin")
+	txn, err = tk.Se.Txn(true)
+	c.Assert(err, IsNil)
+	c.Assert(tk.Se.GetSessionVars().TxnCtx.TxnScope, Equals, oracle.GlobalTxnScope)
+	c.Assert(txn.Valid(), IsTrue)
+	tk.MustExec("insert into t1 (c) values (101)") // write dc-2 with global scope
+	result = tk.MustQuery("select * from t1")      // read dc-1 and dc-2 with global scope
+	c.Assert(len(result.Rows()), Equals, 3)
+	c.Assert(txn.Valid(), IsTrue)
+	tk.MustExec("rollback")
 	result = tk.MustQuery("select * from t1")
+	c.Assert(len(result.Rows()), Equals, 2)
+
+	tk.MustExec("insert into t1 (c) values (101)") // write dc-2 with global scope
+	result = tk.MustQuery("select * from t1")      // read dc-1 and dc-2 with global scope
 	c.Assert(len(result.Rows()), Equals, 3)
 
 	// set txn_scope to local
 	tk.MustExec("set @@session.txn_scope = 'dc-1';")
 	result = tk.MustQuery("select @@txn_scope;")
 	result.Check(testkit.Rows("dc-1"))
+
 	// test local txn
-	tk.MustExec("insert into t1 (c) values (1)") // in dc-1 with dc-1 scope
-	result = tk.MustQuery("select * from t1 where c < 100")
+	tk.MustExec("insert into t1 (c) values (1)")            // write dc-1 with dc-1 scope
+	result = tk.MustQuery("select * from t1 where c < 100") // read dc-1 with dc-1 scope
 	c.Assert(len(result.Rows()), Equals, 3)
+
+	// begin and commit with dc-1 txn scope
 	tk.MustExec("begin")
 	txn, err = tk.Se.Txn(true)
 	c.Assert(err, IsNil)
 	c.Assert(tk.Se.GetSessionVars().TxnCtx.TxnScope, Equals, "dc-1")
 	c.Assert(txn.Valid(), IsTrue)
-	tk.MustExec("insert into t1 (c) values (1)") // in dc-1 with dc-1 scope
-	result = tk.MustQuery("select * from t1 where c < 100")
+	tk.MustExec("insert into t1 (c) values (1)")            // write dc-1 with dc-1 scope
+	result = tk.MustQuery("select * from t1 where c < 100") // read dc-1 with dc-1 scope
 	c.Assert(len(result.Rows()), Equals, 4)
 	c.Assert(txn.Valid(), IsTrue)
 	tk.MustExec("commit")
 	result = tk.MustQuery("select * from t1 where c < 100")
 	c.Assert(len(result.Rows()), Equals, 4)
```

Comment:
These two cases are duplicate. I guess you want to test auto-commit transactions in the first case?

Reply:
Yes, I added a comment to explain this.